README.md

```diff
@@ -1971,4 +1971,244 @@ You can create a `./gradlew checkstyle` command using the following:
+1. Refactor: Optimise your code for readability without changing its behaviour.
+
+First, create a test class that calls the intended target class:
+
+``` java
+public class FibonacciTest {
+    
+    Fibonacci target = new Fibonacci();
+
+}
+```
+
+Then create that target class.
+
+``` java
+public class Fibonacci {
+}
+```
+
+Second, add a simple method to the test class.
+
+``` java
+public class FibonacciTest {
+
+    public void shouldBe() {
+        Fibonacci target = new Fibonacci();
+        target.get(0);
+    }
+
+}
+```
+
+Then create the method `get` in the target class.
+
+``` java
+public class Fibonacci {
+
+    public int get() {
+        return 0;
+    }
+    
+}
+```
+
+Third, create a simple test in the test class.
+
+``` java
+import static org.junit.Assert.assertEquals;
+
+public class FibonacciTest {
+
+    @Test
+    public void shouldBe() {
+        Fibonacci target = new Fibonacci();
+        assertEquals(1,target.get());
+    }
+
+}
+```
+
+Then modify the method in the target class.
+
+``` java
+public class Fibonacci {
+
+    public int get() {
+        return 1;
+    }
+    
+}
+```
+
+Fourth, add more tests/assertions to the test class.
+
+``` java
+import static org.junit.Assert.assertEquals;
+
+public class FibonacciTest {
+
+    @Test
+    public void shouldBe() {
+        Fibonacci target = new Fibonacci();
+        assertEquals(1,target.get());
+        assertEquals(3,target.get());
+    }
+
+}
+```
+
+Then update the target class.
+
+``` java
+public class Fibonacci {
+
+    public int get(int n) {
+        if (n < 3) {
+            return 1;
+        }
+        return 2;
+    }
+}
+```
+
+There is now a lot of refactoring that can be done to the test class.
+
+A target field can be created, as can a private method called `shouldBe()`.
+
+Then further tests/assertions can be added.
+
+``` java
+import static org.junit.Assert.assertEquals;
+
+public class FibonacciTest {
+
+    Fibonacci target = new Fibonacci();
+
+    @Test
+    public void shouldBe1() {
+        shouldBe(1, 1);
+    }
+
+    @Test
+    public void shouldBe2() {
+        shouldBe(2, 1);
+    }
+
+    @Test
+    public void shouldBe3() {
+        shouldBe(3, 2);
+    }
+
+    @Test
+    public void shouldBe4() {
+        shouldBe(4, 3);
+    }
+
+    @Test
+    public void shouldBe5() {
+        shouldBe(5, 5);
+    }
+
+    private void shouldBe(int index, int expected) {
+        assertEquals(expected, target.get(index));
+    }
+}
+```
+
+Finally, update the target class so that it will work for any number tested.
+
+``` java
+public class Fibonacci {
+
+    int get(int n) {
+        if (n < 3) {
+            return 1;
+        }
+        return get(n - 2) + get(n - 1);
+    }
+}
+```
+
+## FizzBuzz
+
+Applicants for junior developer roles are commonly asked to demonstrate on a whiteboard how they would create a class to solve the following problem (sometimes also as a TDD exercise):
```

Comment:
I wouldn't say it's common for junior engineers to be given a TDD test in an interview. That's just for elite students of mine! 😉 
If I were interviewing a junior developer I'd be thrilled if they knew what testing was, and would be well impressed if they could use TDD as well.